catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/AbstractEventController.java

```diff
@@ -101,6 +107,24 @@ public ServerSession getSessionByUserId(String userId) {
     }
 
     /**
+     * Obtains the {@link ServerSession} associated with a given sessionId. Retrieval using
+     * {@link #getSessionByUserId(String)} is the preferred metod. This method is used when
+     * a userId is not available.
+     *
+     * @param sessionId The sessionId associated with the {@code ServerSession} to be retrieved.
+     * @return The {@code ServerSession} associated with the received sessionId or null if the session
+     * does not have an established {@code ServerSession}
+     */
+    public ServerSession getSessionBySessionId(String sessionId) {
```

Comment:
Looking closer, if we need to keep this search, we may want to consider switching over to using a `ConcurrentHashMap` and use `searchValues()` instead.